src/Client/Connection.cpp

```diff
@@ -108,6 +111,17 @@ void Connection::connect(const ConnectionTimeouts & timeouts)
 
         LOG_TRACE(log_wrapper.get(), "Connected to {} server version {}.{}.{}.",
             server_name, server_version_major, server_version_minor, server_version_patch);
+
+        if (server_revision >= DBMS_MIN_REVISION_WITH_INTERSERVER_SECRET && !cluster_secret.empty())
+        {
+#if USE_SSL
+            sendReceiveCluster();
```

Comment:
@vitlibar as you can see that the Cluster package will be sent only if the inter-server secret had been enabled, and if it was ExtendedHello package, you cannot fallback to Hello anyway, since if the inter-server secret had been configured/enabled server cannot ignore this fact

Reply:
I'm not sure about that. On the one hand the server can't ignore the interserver secret but on the other hand what if the server is just older than the client? It would be kind of compatibility breaking change.

1. Interserver secret is disabled => we send Hello request => server sends Hello response

2. Interserver secret is enabled => we send ExtendedHello request => server is new => server sends ExtendedHello response

3a. Interserver secret is enabled => we send ExtendedHello request => server is old => server sends UNEXPECTED_PACKET_FROM_CLIENT => we send Hello request => server sends Hello response

-- OR --

3b. Interserver secret is enabled => we send ExtendedHello request => server is old => server sends UNEXPECTED_PACKET_FROM_CLIENT => we stop because we can't let the server ignore the interserver secret.

Reply:
>I'm not sure about that. On the one hand the server can't ignore the interserver secret but on the other hand what if the server is just older than the client? It would be kind of compatibility breaking change.

Why? One should not enable it if it has clickhouse-server in the cluster w/o this support.

My initial point was, that since you can't ignore the error (in my opinion), there is no much sense in some generic ExtendedHello for this case